lib/editor/tunnel.js

```diff
@@ -248,6 +248,11 @@ class EditorTunnel {
                     })
                 }
             })
+            clearInterval(this.pingInterval)
+            this.pingInterval = setInterval(() => {
+                debug('Sending WS editor tunnel ping')
+                socket.ping()
```

Comment:
Suggest adding a guard here just in case the socket is disconnected when this fires (and the clearTimeouts haven't happen yet) - we do not want an `UncaughtException` killing us here.
```suggestion
                try {
                    socket.ping()
                } catch (err) { }
```